webhooks/util/billing.py

```diff
@@ -0,0 +1,41 @@
+from logging import Logger
+from typing import Any, Dict
+
+from shared.models.endorsement import extract_operation_type_from_endorsement_payload
+
+
+def is_applicable_for_billing(
+    topic: str, payload: Dict[str, Any], logger: Logger
+) -> bool:
+    state = payload.get("state")
+    if topic not in ["proofs", "credentials", "endorsements", "issuer_cred_rev"]:
+        logger.debug(f"Event topic: {topic} is not applicable for the billing service.")
+        return False
+
+    if state not in [
+        "done",
+        "transaction_acked",
+        "revoked",
+        "credential_acked",
+        "presentation_acked",
+    ]:
+        logger.debug(f"Event state: {state} is not applicable for the billing service.")
+        return False
+
+    if topic == "endorsements":
+        operation_type = get_operation_type(payload=payload, logger=logger)
+        if operation_type not in ["1", "100", "102", "113", "114"]:
+            logger.debug(
+                f"Endorsement operation type: {operation_type} is not applicable for the billing service."
```

Comment:
Again, just check all debug logs for f-interpolation please. This one won't be caught with a `logger.debug(f` global search :-) because of the \n